README
FileControl README

Current version: v0.2 (Mar 22th 2013)

DESCRIPTION

FileControl is a file monitoring application. It can monitor
file presense or ebsence, ignore or include folders in scaning,
search with file mask syntax and handle search errors.

Application operate with entries. Each entry include name of
entry, path to search, mask to search (for example: *.txt or 
???file.ext), ignore mask (if search mask = *.txt and ignore
mask = foo.txt then foo.txt will be ignored during scaning),
error condition switch (enabled = treat searching success as
error (files present), disabled = treat searching failure as
error (files ebsent)), case sensetive switch (search with mask
TeSt* will ignore such files as test.file, not work under 
Windows) and folder ignore switch (folders will be ignored).

Application supports several ways to notify user: icon changing
(application icon turn to red even if one of entries return
error) and sound notification. Also FileControl may execute any
command to system shell. Basicly it will launch filemanager to
entry's path with command 'filemanager $PATH'. Command for 
Windows: 'explorer $PATH' or for Linux: 'xdg-open $PATH' or 
even 'mc $PATH'.

CHANGELOG

*See changelog at CHANGELOG file.

DEPENDECIES

libRibbonData - <https://github.com/spoilt-exile/libRibbonData>

LICENSE AND COPYRIGHT

Application licensed under GNU GPLv2+. All source code 
available at <https://github.com/spoilt-exile/FileControl>.

© Stanislav Nepochatov 2013.

THIRD-PARTY COMPONENTS

Copyright for oxygen icons belong to KDE team, original 
copyright file from icons Debian distribution included with
licence file.

Copyright for sound alarm file belong to KDE team, licensed
under GNU GPLv2.

CHANGELOG
==========================================================================================
FileControl v0.1 (Jan 2nd 2013)

Changes:
 ! first release of application;
==========================================================================================
FileControl v0.2 (Mar 22th 2013)

Changes:
 + Add system tray support;
 + Check progress and timestamp display added;
 * Improoved exception handling in ControlTask;

Fixes:
 X Sound alarm fixed (just by using BufferedInputStream);
 X Force control table to repaint after update;
 X Localization fixes;
========================================================================================== 

src/filecontrol/EntryDialog.java
/**
 * This file is part of FileControl application (check README).
 * Copyright (C) 2013  Stanislav Nepochatov
 * 
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 * 
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
**/

package filecontrol;

/**
 * Entry add and edit dialog;
 * @author Stanislav Nepochatov <spoilt.exile@gmail.com>
 */
public class EntryDialog extends javax.swing.JDialog {
    
    /**
     * Null reference indicator;
     */
    private Boolean isNew = false;
    
    /**
     * Processed control entry;
     */
    private ControlEntry processedEntry;
    
    /**
     * Entry which should be return to quene;
     */
    private ControlEntry returnedEntry;

    /**
     * Creates new form EntryDialog
     */
    public EntryDialog(java.awt.Frame parent, boolean modal, ControlEntry givenEntry) {
        super(parent, modal);
        initComponents();
        if (givenEntry == null) {
            this.saveBut.setText("Добавить");
            this.isNew = true;
        } else {
            this.processedEntry = givenEntry;
            this.entryNameField.setText(this.processedEntry.EntryName);
            this.pathField.setText(this.processedEntry.Path);
            this.maskField.setText(this.processedEntry.Mask);
            this.ignoreMaskField.setText(this.processedEntry.IgnoreMask);
            this.caseSwitch.setSelected(this.processedEntry.CaseSensetive);
            this.errorCondSwitch.setSelected(this.processedEntry.ErrorCondition);
            this.folderIgnore.setSelected(this.processedEntry.IgnoreFolder);
        }
    }

    /**
     * This method is called from within the constructor to initialize the form. WARNING: Do NOT modify this code. The content of this method is always
     * regenerated by the Form Editor.
     */
    @SuppressWarnings("unchecked")
    // <editor-fold defaultstate="collapsed" desc="Generated Code">//GEN-BEGIN:initComponents
    private void initComponents() {

        entryNameField = new javax.swing.JTextField();
        jLabel1 = new javax.swing.JLabel();
        pathField = new javax.swing.JTextField();
        jLabel2 = new javax.swing.JLabel();
        maskField = new javax.swing.JTextField();
        jLabel3 = new javax.swing.JLabel();
        ignoreMaskField = new javax.swing.JTextField();
        jLabel4 = new javax.swing.JLabel();
        errorCondSwitch = new javax.swing.JCheckBox();
        caseSwitch = new javax.swing.JCheckBox();
        cancelBut = new javax.swing.JButton();
        saveBut = new javax.swing.JButton();
        folderIgnore = new javax.swing.JCheckBox();

        setDefaultCloseOperation(javax.swing.WindowConstants.DISPOSE_ON_CLOSE);
        setTitle("Настройка записи");

        jLabel1.setText("Имя записи");

        jLabel2.setText("Путь поиска");

        jLabel3.setText("Маска");

        jLabel4.setText("Игнорировать маску");

        errorCondSwitch.setSelected(true);
        errorCondSwitch.setText("Сигнализировать о присутствии");

        caseSwitch.setText("Учитывать регистр");

        cancelBut.setIcon(new javax.swing.ImageIcon(getClass().getResource("/icons/cancel.png"))); // NOI18N
        cancelBut.setText("Отмена");
        cancelBut.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                cancelButActionPerformed(evt);
            }
        });

        saveBut.setIcon(new javax.swing.ImageIcon(getClass().getResource("/icons/save.png"))); // NOI18N
        saveBut.setText("Сохранить");
        saveBut.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                saveButActionPerformed(evt);
            }
        });

        folderIgnore.setSelected(true);
        folderIgnore.setText("Игнорировать папки ");

        javax.swing.GroupLayout layout = new javax.swing.GroupLayout(getContentPane());
        getContentPane().setLayout(layout);
        layout.setHorizontalGroup(
            layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(layout.createSequentialGroup()
                .addGap(25, 25, 25)
                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                    .addGroup(layout.createSequentialGroup()
                        .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                            .addComponent(jLabel1, javax.swing.GroupLayout.Alignment.TRAILING)
                            .addComponent(jLabel2, javax.swing.GroupLayout.Alignment.TRAILING)
                            .addComponent(jLabel3, javax.swing.GroupLayout.Alignment.TRAILING)
                            .addComponent(jLabel4, javax.swing.GroupLayout.Alignment.TRAILING))
                        .addGap(18, 18, 18)
                        .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                            .addComponent(entryNameField, javax.swing.GroupLayout.DEFAULT_SIZE, 225, Short.MAX_VALUE)
                            .addGroup(layout.createSequentialGroup()
                                .addComponent(pathField)
                                .addGap(65, 65, 65))
                            .addComponent(maskField)
                            .addComponent(ignoreMaskField)))
                    .addGroup(javax.swing.GroupLayout.Alignment.TRAILING, layout.createSequentialGroup()
                        .addComponent(saveBut)
                        .addGap(18, 18, 18)
                        .addComponent(cancelBut))
                    .addGroup(layout.createSequentialGroup()
                        .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                            .addComponent(folderIgnore)
                            .addComponent(errorCondSwitch)
                            .addComponent(caseSwitch))
                        .addGap(0, 0, Short.MAX_VALUE)))
                .addContainerGap())
        );
        layout.setVerticalGroup(
            layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(layout.createSequentialGroup()
                .addContainerGap()
                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.BASELINE)
                    .addComponent(entryNameField, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
                    .addComponent(jLabel1))
                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.UNRELATED)
                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.BASELINE)
                    .addComponent(pathField, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
                    .addComponent(jLabel2))
                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.UNRELATED)
                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.BASELINE)
                    .addComponent(maskField, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
                    .addComponent(jLabel3))
                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.UNRELATED)
                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.BASELINE)
                    .addComponent(ignoreMaskField, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
                    .addComponent(jLabel4))
                .addGap(18, 18, 18)
                .addComponent(errorCondSwitch)
                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.UNRELATED)
                .addComponent(caseSwitch)
                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.UNRELATED)
                .addComponent(folderIgnore)
                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED, 17, Short.MAX_VALUE)
                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.BASELINE)
                    .addComponent(cancelBut)
                    .addComponent(saveBut))
                .addContainerGap())
        );

        pack();
    }// </editor-fold>//GEN-END:initComponents

    private void cancelButActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_cancelButActionPerformed
        this.dispose();
    }//GEN-LAST:event_cancelButActionPerformed

    private void saveButActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_saveButActionPerformed
        this.returnedEntry = new ControlEntry(this.entryNameField.getText(),
            this.pathField.getText(), this.maskField.getText(), 
            this.ignoreMaskField.getText(), this.caseSwitch.isSelected(),
            this.errorCondSwitch.isSelected(), this.folderIgnore.isSelected());
        if (this.isNew) {
            ControlQuene.addToQuene(returnedEntry);
        } else {
            ControlQuene.replaceWithinQuene(processedEntry, returnedEntry);
        }
        FileControl.MainWindow.buildControlTable();
        this.dispose();
    }//GEN-LAST:event_saveButActionPerformed

    /**
     * @param args the command line arguments
     */
    public static void main(String args[]) {
        /*
         * Set the Nimbus look and feel
         */
        //<editor-fold defaultstate="collapsed" desc=" Look and feel setting code (optional) ">
        /*
         * If Nimbus (introduced in Java SE 6) is not available, stay with the default look and feel. For details see
         * http://download.oracle.com/javase/tutorial/uiswing/lookandfeel/plaf.html
         */
        try {
            for (javax.swing.UIManager.LookAndFeelInfo info : javax.swing.UIManager.getInstalledLookAndFeels()) {
                if ("Nimbus".equals(info.getName())) {
                    javax.swing.UIManager.setLookAndFeel(info.getClassName());
                    break;
                }
            }
        } catch (ClassNotFoundException ex) {
            java.util.logging.Logger.getLogger(EntryDialog.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (InstantiationException ex) {
            java.util.logging.Logger.getLogger(EntryDialog.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (IllegalAccessException ex) {
            java.util.logging.Logger.getLogger(EntryDialog.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (javax.swing.UnsupportedLookAndFeelException ex) {
            java.util.logging.Logger.getLogger(EntryDialog.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        }
        //</editor-fold>

        /*
         * Create and display the dialog
         */
        java.awt.EventQueue.invokeLater(new Runnable() {

            public void run() {
                EntryDialog dialog = new EntryDialog(new javax.swing.JFrame(), true, null);
                dialog.addWindowListener(new java.awt.event.WindowAdapter() {

                    @Override
                    public void windowClosing(java.awt.event.WindowEvent e) {
                        System.exit(0);
                    }
                });
                dialog.setVisible(true);
            }
        });
    }
    // Variables declaration - do not modify//GEN-BEGIN:variables
    private javax.swing.JButton cancelBut;
    private javax.swing.JCheckBox caseSwitch;
    private javax.swing.JTextField entryNameField;
    private javax.swing.JCheckBox errorCondSwitch;
    private javax.swing.JCheckBox folderIgnore;
    private javax.swing.JTextField ignoreMaskField;
    private javax.swing.JLabel jLabel1;
    private javax.swing.JLabel jLabel2;
    private javax.swing.JLabel jLabel3;
    private javax.swing.JLabel jLabel4;
    private javax.swing.JTextField maskField;
    private javax.swing.JTextField pathField;
    private javax.swing.JButton saveBut;
    // End of variables declaration//GEN-END:variables
}


src/filecontrol/EntryStatus.java
/**
 * This file is part of FileControl application (check README).
 * Copyright (C) 2013  Stanislav Nepochatov
 * 
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 * 
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
**/

package filecontrol;

/**
 * Control entry status.
 * @author Stanislav Nepochatov
 */
public enum EntryStatus {
    
    /**
     * All is fine;
     */
    OK,
    
    /**
     * IOException recieved while scaning;
     */
    IOEXCEPTION_OCURRED,
    
    /**
     * Condition has been satisfied;
     */
    CONDITION_SATISFIED
}


src/filecontrol/SettingsDialog.java
/**
 * This file is part of FileControl application (check README).
 * Copyright (C) 2013  Stanislav Nepochatov
 * 
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 * 
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
**/

package filecontrol;

/**
 * Control general settings dialog;
 * @author s.nepochatov
 */
public class SettingsDialog extends javax.swing.JDialog {

    /**
     * Creates new form SettingsDialog
     */
    public SettingsDialog(java.awt.Frame parent, boolean modal) {
        super(parent, modal);
        initComponents();
        this.openFileManagerFlag.setSelected(FileControl.MainProperties.getProperty("err_open_filemanager").equals("1") ? true : false);
        this.fileManagerCommandField.setText(FileControl.MainProperties.getProperty("err_file_handler"));
        this.soundAlarmFlag.setSelected(FileControl.MainProperties.getProperty("err_play_sound").equals("1") ? true : false);
    }

    /**
     * This method is called from within the constructor to initialize the form.
     * WARNING: Do NOT modify this code. The content of this method is always
     * regenerated by the Form Editor.
     */
    @SuppressWarnings("unchecked")
    // <editor-fold defaultstate="collapsed" desc="Generated Code">//GEN-BEGIN:initComponents
    private void initComponents() {
        bindingGroup = new org.jdesktop.beansbinding.BindingGroup();

        openFileManagerFlag = new javax.swing.JCheckBox();
        fileManagerCommandField = new javax.swing.JTextField();
        jLabel1 = new javax.swing.JLabel();
        soundAlarmFlag = new javax.swing.JCheckBox();
        cancelFlag = new javax.swing.JButton();
        saveBut = new javax.swing.JButton();

        setDefaultCloseOperation(javax.swing.WindowConstants.DISPOSE_ON_CLOSE);
        setTitle("Настройки");

        openFileManagerFlag.setSelected(true);
        openFileManagerFlag.setText("Открывать файлменеджер");

        fileManagerCommandField.setText("explorer $PATH");

        org.jdesktop.beansbinding.Binding binding = org.jdesktop.beansbinding.Bindings.createAutoBinding(org.jdesktop.beansbinding.AutoBinding.UpdateStrategy.READ_WRITE, openFileManagerFlag, org.jdesktop.beansbinding.ELProperty.create("${selected}"), fileManagerCommandField, org.jdesktop.beansbinding.BeanProperty.create("enabled"));
        bindingGroup.addBinding(binding);

        jLabel1.setText("Команда");

        soundAlarmFlag.setSelected(true);
        soundAlarmFlag.setText("Звуковой сигнал");

        cancelFlag.setIcon(new javax.swing.ImageIcon(getClass().getResource("/icons/cancel.png"))); // NOI18N
        cancelFlag.setText("Отмена");
        cancelFlag.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                cancelFlagActionPerformed(evt);
            }
        });

        saveBut.setIcon(new javax.swing.ImageIcon(getClass().getResource("/icons/save.png"))); // NOI18N
        saveBut.setText("Сохранить");
        saveBut.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                saveButActionPerformed(evt);
            }
        });

        javax.swing.GroupLayout layout = new javax.swing.GroupLayout(getContentPane());
        getContentPane().setLayout(layout);
        layout.setHorizontalGroup(
            layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(layout.createSequentialGroup()
                .addContainerGap()
                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                    .addGroup(layout.createSequentialGroup()
                        .addComponent(jLabel1)
                        .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.UNRELATED)
                        .addComponent(fileManagerCommandField))
                    .addGroup(javax.swing.GroupLayout.Alignment.TRAILING, layout.createSequentialGroup()
                        .addGap(0, 0, Short.MAX_VALUE)
                        .addComponent(saveBut)
                        .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.UNRELATED)
                        .addComponent(cancelFlag))
                    .addGroup(layout.createSequentialGroup()
                        .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                            .addComponent(openFileManagerFlag)
                            .addComponent(soundAlarmFlag))
                        .addGap(0, 74, Short.MAX_VALUE)))
                .addContainerGap())
        );
        layout.setVerticalGroup(
            layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(layout.createSequentialGroup()
                .addContainerGap()
                .addComponent(openFileManagerFlag)
                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.UNRELATED)
                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.BASELINE)
                    .addComponent(fileManagerCommandField, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
                    .addComponent(jLabel1))
                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.UNRELATED)
                .addComponent(soundAlarmFlag)
                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED, 14, Short.MAX_VALUE)
                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.BASELINE)
                    .addComponent(cancelFlag)
                    .addComponent(saveBut))
                .addContainerGap())
        );

        bindingGroup.bind();

        pack();
    }// </editor-fold>//GEN-END:initComponents

    private void saveButActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_saveButActionPerformed
        FileControl.MainProperties.setProperty("err_open_filemanager", this.openFileManagerFlag.isSelected() ? "1" : "0");
        FileControl.MainProperties.setProperty("err_file_handler", this.fileManagerCommandField.getText());
        FileControl.MainProperties.setProperty("err_play_sound", this.soundAlarmFlag.isSelected() ? "1" : "0");
        FileControl.storeProperties();
        this.dispose();
    }//GEN-LAST:event_saveButActionPerformed

    private void cancelFlagActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_cancelFlagActionPerformed
        this.dispose();
    }//GEN-LAST:event_cancelFlagActionPerformed

    /**
     * @param args the command line arguments
     */
    public static void main(String args[]) {
        /*
         * Set the Nimbus look and feel
         */
        //<editor-fold defaultstate="collapsed" desc=" Look and feel setting code (optional) ">
        /*
         * If Nimbus (introduced in Java SE 6) is not available, stay with the
         * default look and feel. For details see
         * http://download.oracle.com/javase/tutorial/uiswing/lookandfeel/plaf.html
         */
        try {
            for (javax.swing.UIManager.LookAndFeelInfo info : javax.swing.UIManager.getInstalledLookAndFeels()) {
                if ("Nimbus".equals(info.getName())) {
                    javax.swing.UIManager.setLookAndFeel(info.getClassName());
                    break;
                }
            }
        } catch (ClassNotFoundException ex) {
            java.util.logging.Logger.getLogger(SettingsDialog.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (InstantiationException ex) {
            java.util.logging.Logger.getLogger(SettingsDialog.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (IllegalAccessException ex) {
            java.util.logging.Logger.getLogger(SettingsDialog.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (javax.swing.UnsupportedLookAndFeelException ex) {
            java.util.logging.Logger.getLogger(SettingsDialog.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        }
        //</editor-fold>

        /*
         * Create and display the dialog
         */
        java.awt.EventQueue.invokeLater(new Runnable() {

            public void run() {
                SettingsDialog dialog = new SettingsDialog(new javax.swing.JFrame(), true);
                dialog.addWindowListener(new java.awt.event.WindowAdapter() {

                    @Override
                    public void windowClosing(java.awt.event.WindowEvent e) {
                        System.exit(0);
                    }
                });
                dialog.setVisible(true);
            }
        });
    }
    // Variables declaration - do not modify//GEN-BEGIN:variables
    private javax.swing.JButton cancelFlag;
    private javax.swing.JTextField fileManagerCommandField;
    private javax.swing.JLabel jLabel1;
    private javax.swing.JCheckBox openFileManagerFlag;
    private javax.swing.JButton saveBut;
    private javax.swing.JCheckBox soundAlarmFlag;
    private org.jdesktop.beansbinding.BindingGroup bindingGroup;
    // End of variables declaration//GEN-END:variables
}
